kubesae/ansible/deploy.py

```diff
@@ -46,8 +68,14 @@ def ansible_playbook(c, name, extra="", verbosity=1):
 
     Usage: inv deploy.playbook <PLAYBOOK>.yaml --extra=<EXTRA> --verbosity=<VERBOSITY>
     """
+    if c.config.get("aws") and c.config.aws.get("profile_name"):
+        # if we're using AWS and using an AWS_PROFILE, then we'll adjust the shell
+        # environment for boto's sake
+        shell_env = get_boto_env(c.config.aws.get("profile_name"))
+    else:
+        shell_env = {}
     with c.cd("deploy/"):
-        c.run(f"ansible-playbook {name} {extra} -{'v'*verbosity}")
+        c.run(f"ansible-playbook {name} {extra} -{'v'*verbosity}", env=shell_env)
```

Comment:
Neat, looks like env [updates the default environment dict](http://docs.pyinvoke.org/en/stable/api/runners.html#invoke.runners.Runner.run) by default, rather than replace it. 🆒